administrative-sdk/choiceRecognition.js

```diff
@@ -273,91 +245,68 @@ class ChoiceRecognition {
     if (!challenge || !challenge.id) {
-      throw new Error('challenge.id field is required');
+      return Promise.reject(new Error('challenge.id field is required'));
     }
     if (!challenge.organisationId) {
-      throw new Error('challenge.organisationId field is required');
+      return Promise.reject(new Error('challenge.organisationId field is required'));
     }
-
     var url = connection.settings.apiUrl + '/organisations/' +
       challenge.organisationId + '/challenges/choice/' +
       challenge.id + '/recognitions/' + recognitionId;
-    connection._secureAjaxGet(url, _cb, ecb);
-  }
 
-  /**
-   * Callback used by listChoiceRecognitions.
-   *
-   * @callback Sdk~listChoiceRecognitionsCallback
-   * @param {its.ChoiceRecognition[]} recognitions Retrieved choice recognition domain model instances.
-   */
-  listChoiceRecognitionCallback(recognitions) {}
-
-  /**
-   * Error callback used by listChoiceRecognitions.
-   *
-   * @callback Sdk~listChoiceRecognitionsErrorCallback
-   * @param {object[]} errors Array of errors.
-   */
-  listChoiceRecognitionErrorCallback(errors) {}
-
-  /**
-   * List all choice recognitions in a specific choice challenge.
-   *
-   * @param {ChoiceChallenge} challenge Specify a choice challenge to list speech recognitions for.
-   * @param {Sdk~listChoiceRecognitionsCallback} cb The callback that handles the response.
-   * @param {Sdk~listChoiceRecognitionsErrorCallback} [ecb] The callback that handles the error response.
-   */
-  static listChoiceRecognitions(connection, challenge, cb, ecb) {
-    var _cb = function(data) {
-      var recognitions = [];
-      data.forEach(function(datum) {
+    return connection._secureAjaxGet(url)
+      .then(datum => {
         var student = new Student(challenge.organisationId, datum.studentId);
         var recognition = new ChoiceRecognition(challenge, student,
           datum.id, new Date(datum.created), new Date(datum.updated),
           datum.audioUrl);
-        // Recognition may not be successful, in which case the recognition
+        // Alignment may not be successful, in which case the recognition
         // is not available, but it's still an attempt that is available,
-        // albeit without extended attributes like recognised.
+        // albeit without extended attributes like score and phonemes.
         if (datum.recognised) {
           recognition.recognised = datum.recognised;
         }
-        recognitions.push(recognition);
+        return recognition;
       });
-      if (cb) {
-        cb(recognitions);
-      }
-    };
+  }
 
+  /**
+   * List all choice recognitions in a specific choice challenge.
+   *
+   * @param {ChoiceChallenge} challenge Specify a choice challenge to list speech recognitions for.
+   * @param {Sdk~listChoiceRecognitionsCallback} cb The callback that handles the response.
+   * @param {Sdk~listChoiceRecognitionsErrorCallback} [ecb] The callback that handles the error response.
```

Comment:
Missing documentation update. Callbacks are no longer taken and a promise is returned.